xarray/coding/cftime_offsets.py

```diff
@@ -672,22 +673,22 @@ def to_offset(freq):
 
 
 def to_cftime_datetime(date_str_or_date, calendar=None):
-    import cftime
+    cftime_datetime_base = _import_cftime_datetime_base()
 
     if isinstance(date_str_or_date, str):
         if calendar is None:
             raise ValueError(
-                "If converting a string to a cftime.datetime object, "
+                "If converting a string to a cftime datetime object, "
                 "a calendar type must be provided"
             )
         date, _ = _parse_iso8601_with_reso(get_date_type(calendar), date_str_or_date)
         return date
-    elif isinstance(date_str_or_date, cftime.datetime):
+    elif isinstance(date_str_or_date, cftime_datetime_base):
         return date_str_or_date
     else:
         raise TypeError(
             "date_str_or_date must be a string or a "
-            "subclass of cftime.datetime. Instead got "
+            "subclass of cftime.datetime_base. Instead got "
```

Comment:
Do you need to be more explicit here and also mention `cftime.datetime`? Maybe
```suggestion
            "subclass of cftime.datetime_base (formerly cftime.datetime). Instead got "
```
?